example/postconstructorpredestructor/app.d

```diff
@@ -42,8 +42,14 @@ class AClass {
 
 public void main() {
 	auto container = new shared DependencyContainer();
-	container.register!ADependency;
-	container.register!AClass;
+	container.register!(ADependency).onConstructed((Object obj) {
+		writeln("ADependency constructed");
+	});
+
+	container.register!(AClass).onConstructed((Object obj) {
+		writeln("AClass constructed");
+	});
+

```

Comment:
Could also be put in its own example. Examples try to showcase one thing only to teach people topic-by-topic.